src/Orleans.Runtime/Versions/GrainVersionStore.cs

```diff
@@ -14,21 +17,13 @@ internal class GrainVersionStore : IVersionStore
         private readonly string clusterId;
         private IVersionStoreGrain StoreGrain => this.grainFactory.GetGrain<IVersionStoreGrain>(this.clusterId);
 
-        public bool IsEnabled { get; private set; }
+        public bool IsEnabled { get; }
 
-        public GrainVersionStore(IInternalGrainFactory grainFactory, ILocalSiloDetails siloDetails)
+        public GrainVersionStore(IInternalGrainFactory grainFactory, ILocalSiloDetails siloDetails, IServiceProvider services)
         {
             this.grainFactory = grainFactory;
             this.clusterId = siloDetails.ClusterId;
-            this.IsEnabled = false;
-        }
-
-        public void SetStorageManager(IStorageProviderManager storageProviderManager)
-        {
-            IStorageProvider unused;
-            IsEnabled = storageProviderManager.TryGetProvider(
-                ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME,
-                out unused);
+            this.IsEnabled = services.GetService<IStorageProvider>() != null;
```

Comment:
I'm not actually sure if it's fine for a factory to return `null` - I'm not seeing it in the DI specification tests https://github.com/aspnet/DependencyInjection/blob/dev/src/DI.Specification.Tests/DependencyInjectionSpecificationTests.cs

It might be worth quickly clarifying, or alternatively changing the code which registers this default provider to only register it if a keyed service with that name exists.

Reply:
Other places in our code make this same assumption, and we use GetRequiredService when null is not valid.  If this is not safe we'll need to open an issue and address all instances of this.  I suggest we leave this logic as is, and loop back on the possible improper use of get service in a future work item.

Reply:
Sounds fine to me, it's not a major concern: any issues here will blow up the silo on startup anyway.